### DISCUSSION TITLE
Mandatory utm_source parameter

### DISCUSSION THREAD

**Discussion by wiesson** [plausible/analytics#1277] [Anything Else]
From the docs: https://plausible.io/blog/utm-tracking-tags#types-of-utm-parameters
> The utm_source parameter is mandatory and required. Make sure to use it in all UTM tagged links.

I have a promo-bar on my website and I'd like to add a campaign tag auch as `?utm_campaign=promo-september` to quickly identify where the traffic is coming from (E.g. from the promo bar, or some other listing on the same page). Do I need the `utm_source` here as well? If yes, what value is appropriate if the source is the same domain / page and not external? 

Does it even make sense to use utm tags here? If the first utm tag starts the session, (coming from instagram), then clicks on my promo-bar, I guess it's not counted anyway?

**Comment by metmarkosaric** [comment]
hi @wiesson! utm tags are meant for external traffic. it is recommended not to use them for internal traffic. to track internally, it is best to use [our custom events](https://plausible.io/docs/custom-event-goals).

**Reply by wiesson** [reply]
Thanks! And do I need `utm_source` for instagram posts? Or is it ok to use `utm_campaign` only?

**Comment by metmarkosaric** [comment]
utm_source is required. you could shorten your UTM tagged link with a URL shortener to make it look nicer in social media if that's your concern

if you'd like to minimize the tags used as much as possible you could also use ?ref=<value> or ?source=<value> instead of UTM tags

**Reply by wiesson** [reply]
Thanks! My question has been resolved :)

**Reply by metmarkosaric** [reply]
you're welcome! have fun!
